warehouse/accounts/services.py

```diff
@@ -180,61 +180,30 @@ def verify_email(self, user_id, email_address):
                 email.verified = True
 
 
-@implementer(IUserTokenService)
-class UserTokenService:
-    def __init__(self, user_service, settings):
-        self.user_service = user_service
-        self.serializer = URLSafeTimedSerializer(
-            settings["password_reset.secret"],
-            salt="password-reset",
-        )
-        self.token_max_age = settings["password_reset.token_max_age"]
+@implementer(ITokenService)
+class TokenService:
+    def __init__(self, secret, salt, max_age):
+        self.serializer = URLSafeTimedSerializer(secret, salt=salt)
+        self.max_age = max_age
 
-    def generate_token(self, user):
+    def generate_token(self, data):
         return self.serializer.dumps({
-            "user.id": str(user.id),
-            "user.last_login": str(user.last_login),
-            "user.password_date": str(user.password_date),
+            key: str(value)
+            for key, value in data.items()
         })
 
-    def get_user_by_token(self, token):
+    def extract_data(self, token):
```

Comment:
The names `generate_token` and `extract_data` feel kinda unharmonious to me.

The names `dumps` and `loads` from its dangerous itself are generic enough for our purposes, particularly on the generalized service. This is mostly a nit and not a blocker.